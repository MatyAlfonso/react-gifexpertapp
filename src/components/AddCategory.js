import React, { useState } from 'react';
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length > 1) {
            setCategories(cats => [inputValue, ...cats]);
            setInputValue('');
        }

    }

    return (
        <form className="add-category__layout" onSubmit={handleSubmit}>
            <input
                placeholder="Search some GIF..."
                type="text"
                value={inputValue}
                onChange={handleInputChange}
                className="add-category__input"
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

