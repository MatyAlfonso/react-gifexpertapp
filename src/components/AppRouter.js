import React from 'react';
import {
    HashRouter as Router,
    Routes,
    Route
} from 'react-router-dom';

import { SearchPage } from '../routes/SearchPage';
import { MainPage } from '../routes/MainPage';

export const AppRouter = () => {
    return (
        <Router>
            <Routes>
                <Route path="/" element={<MainPage />} />
                <Route path="/search" element={<SearchPage />} />
            </Routes>
        </Router>
    )
}
