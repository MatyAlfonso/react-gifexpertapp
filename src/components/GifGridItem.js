import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ( {title, url} ) => {
    //console.log(id, title, url);
    return (
        <div className="gif-grid-item__card animate__animated animate__fadeIn">
            <img className="gif-grid-item__card-img" src={url} alt={title} />
            <p className="gif-grid-item__card-text">{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title : PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
