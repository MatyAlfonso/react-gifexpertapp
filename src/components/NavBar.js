//import { useState } from "react";
import { Link } from "react-router-dom";

//import hamburguerMenu from "../styles/images/hamburguer-menu.png";
//import closeMenu from "../styles/images/go-back.png";
import logo from "../styles/images/the-gif-searcher-logo.png";

export const NavBar = () => {

    //const [navBarState, setNavBarState] = useState(false);

    /*let handleClick = () => {
        setNavBarState(!navBarState);
        navBarState ? console.log('Menu cerrado') : console.log('Menu abierto');
    }*/


    return (
        <div className="nav-bar__layout">
            {/*navBarState ?
                <nav className="nav-bar__body">
                    <img onClick={() => handleClick()} className="nav-bar__hamburguer-menu-opened" src={closeMenu} alt="Close menu" />
                    <div className="nav-bar__container">
                        <Link to="/">
                            <button className="nav-bar__button">Home</button>
                        </Link>
                        <Link to="search">
                            <button className="nav-bar__button">Search Gifs</button>
                        </Link>
                        <Link to="/">
                            <button className="nav-bar__button">Last Search</button>
                        </Link>
                    </div>
                </nav>
                :
                <nav>
                    <img onClick={() => handleClick()} className="nav-bar__hamburguer-menu-closed" src={hamburguerMenu} alt="Hamburguer menu" />
                </nav>
            */}
            <nav className="nav-bar__container">
                <Link to="/">
                    <img className="nav-bar__logo" src={logo} alt="The Gif Searcher Logo" />
                </Link>
            </nav>
        </div>
    )
}
