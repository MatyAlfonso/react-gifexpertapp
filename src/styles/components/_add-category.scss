.add-category__layout{
    display: flex;
    justify-content: center;
}

.add-category__input{
    background-color: $lightBlueColor;
    border: none;
    border-radius: 10px;
    height: 30px;
    width: 300px;
    color: $whiteColor;
    font-weight: bold;
    box-shadow: 0 2px 1px #888888;
    outline: none;
    text-align: center;
    display: block;
}

::-webkit-input-placeholder { /* WebKit, Blink, Edge */
    color: $whiteColor;
    font-weight: 400;
}

@media (min-width: 700px){
    .add-category__input{
        height: 50px;
        width: 300px;
        font-size: 1em;
    }
}