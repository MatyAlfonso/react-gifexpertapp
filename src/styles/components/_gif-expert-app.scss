.gif-expert-app__main{
    display: flex;
    flex-direction: column;
}

.gif-expert-app__we-can-found-logo{
    margin: 10% auto;
    height: auto;
    width: 300px;
    padding: 10% 0 10% 0;
}

.gif-expert-app__back-to-top-btn{
    position: fixed;
    bottom: 20px;
    right: 20px;

    &:hover{
        transform: scale(1.1,1.1);
    }
}

@media (min-width: 700px){
    .gif-expert-app__we-can-found-logo{
        width: 350px;
        padding: 8% 0 4% 0;
    }
}

@media (min-width: 1024px){
    .gif-expert-app__we-can-found-logo{
        margin: 6% auto;
        width: 350px;
        padding: 8% 0 2% 0;
    }
}

@media (min-width: 1440px){
    .gif-expert-app__we-can-found-logo{
        margin: 2% auto;
        width: 400px;
        padding: 8% 0 2% 0;
    }
}