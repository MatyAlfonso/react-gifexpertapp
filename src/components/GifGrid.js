import React from 'react'
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const { data : images, loading } = useFetchGifs(category);
    

    return (
        <>
            <h3 className="gif-grid__center animate__animated animate__zoomIn">{`Results for "${category}"`}</h3>
            {loading && <p className="animate__animated animate__flash">Loading...</p>}
            <div className="gif-grid__card-grid">

                {
                    images.map((img) => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category : PropTypes.string.isRequired
}
