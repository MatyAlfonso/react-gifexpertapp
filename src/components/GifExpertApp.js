import React, { useState } from 'react';

import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

import weCanFoundLogo from '../styles/images/we-can-found.png';
import goToTop from '../styles/images/go-to-top.png';

export const GifExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState(defaultCategories); //Hook

    /*const handleAdd = () => {
        //setCategories(categories => [...categories, 'Spiderman']);
        setCategories([...categories, 'Spiderman']);
    }*/

    const handleClick = () => {
        window.scrollTo(0, 0,);
    }

    return (
        <div id="top" className="gif-expert-app__main">
            <img className="gif-expert-app__we-can-found-logo" src={weCanFoundLogo} alt="We can found the gif you were looking for - logo" />
            <AddCategory setCategories={setCategories} />

            <ol>
                {
                    categories.map((category) => (
                        <GifGrid
                            key={category}
                            category={category}
                        />
                    ))
                }
            </ol>

            <img
                onClick={() => handleClick()}
                className="gif-expert-app__back-to-top-btn"
                src={goToTop}
                alt="Button to return to the top of the page"
            />


        </div>
    )
}
