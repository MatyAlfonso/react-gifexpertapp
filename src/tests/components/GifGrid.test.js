import React from 'react';
import '@testing-library/jest-dom';
import { GifGrid } from '../../components/GifGrid';
import { shallow } from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {
    const category = 'One Punch';

    test('Debe mostrarse correctamente.', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar items cuando se cargan imÃ¡genes con useFetchGifs.', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://google.com',
            title: 'ABC Title'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    });
})