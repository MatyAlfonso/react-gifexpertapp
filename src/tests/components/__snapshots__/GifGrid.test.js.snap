// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe mostrar items cuando se cargan im√°genes con useFetchGifs. 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__zoomIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="ABC Title"
      url="https://google.com"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid /> Debe mostrarse correctamente. 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__zoomIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
